HADOOP-18087. fix bugs when looking up record from upstream DNS servers.

eubnara opened the pull request:
<!--
  Thanks for sending a pull request!
    1. If this is your first time, please read our contributor guidelines: https://cwiki.apache.org/confluence/display/HADOOP/How+To+Contribute
    2. Make sure your PR title starts with JIRA issue id, e.g., 'HADOOP-17799. Your PR title ...'.
-->

### Description of PR

When query A record which is chained by CNAME, YARN Registry DNS Server does not properly respond. Some CNAME records are missing.

For example, "repo.maven.apache.org" is chaned as follows:

repo.maven.apache.org.	21317	IN	CNAME	repo.apache.maven.org.
repo.apache.maven.org.	20114	IN	CNAME	maven.map.fastly.net.
maven.map.fastly.net.	7	IN	A	199.232.192.215
maven.map.fastly.net.	7	IN	A	199.232.196.215

If ask A record for "repo.maven.apache.org" using "dig" or "nslookup", YARN Registry DNS Server will give answers similar to this:
(10.1.2.3, 10.8.8.8 IP is virtual)


```
$ nslookup repo.maven.apache.org 10.1.2.3
Server:		10.1.2.3
Address:	10.1.2.3#53

Non-authoritative answer:
repo.maven.apache.org	canonical name = repo.apache.maven.org.
Name:	maven.map.fastly.net
Address: 151.101.196.215
** server can't find repo.apache.maven.org: NXDOMAIN
```

The reason why you can see "NXDOMAIN", "nslookup" will query "A" & "AAAA" records.
If there is no answer from other dns server, "answers == null" but YARN Registry DNS Server has a bug. There is no null handling.


```java
    // Forward lookup to primary DNS servers
    Record[] answers = getRecords(name, type);
    try {
      for (Record r : answers) {
        if (!response.findRecord(r)) {
          if (r.getType() == Type.SOA) {
            response.addRecord(r, Section.AUTHORITY);
          } else {
            response.addRecord(r, Section.ANSWER);
          }
        }
        if (r.getType() == Type.CNAME) {
          Name cname = r.getName();
          if (iterations < 6) {
            remoteLookup(response, cname, type, iterations + 1);
          }
        }
      }
    } catch (NullPointerException e) {
      return Rcode.NXDOMAIN;
    } catch (Throwable e) {
      return Rcode.SERVFAIL;
    }
    return Rcode.NOERROR;
```





It should be like this:

```
nslookup repo.maven.apache.org 10.8.8.8
Server:		10.8.8.8
Address:	10.8.8.8#53

Non-authoritative answer:
repo.maven.apache.org	canonical name = repo.apache.maven.org.
repo.apache.maven.org	canonical name = maven.map.fastly.net.
Name:	maven.map.fastly.net
Address: 151.101.196.215
```



### How was this patch tested?

- apply this patch to YARN Registry DNS Server in our cluster & test with `dig` and `nslookup`

### For code changes:

- [x] Does the title or this PR starts with the corresponding JIRA issue id (e.g. 'HADOOP-17799. Your PR title ...')?
- [ ] Object storage: have the integration tests been executed and the endpoint declared according to the connector-specific documentation?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the `LICENSE`, `LICENSE-binary`, `NOTICE-binary` files?

eubnara committed:
HADOOP-18087. fix bugs when looking up record from upstream DNS servers.

- add chained CNAME records to answer section
- distinguish between NXDOMAIN and NOERROR + empty answer

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 57s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  35m 13s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  24m 15s |  |  trunk passed with JDK Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04  |
| +1 :green_heart: |  compile  |  20m 26s |  |  trunk passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  checkstyle  |   0m 30s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 38s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   0m 36s |  |  trunk passed with JDK Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04  |
| +1 :green_heart: |  javadoc  |   0m 33s |  |  trunk passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  spotbugs  |   0m 54s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  23m 18s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 20s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  23m 27s |  |  the patch passed with JDK Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04  |
| +1 :green_heart: |  javac  |  23m 27s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  20m 34s |  |  the patch passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  javac  |  20m 34s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   0m 28s |  |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   0m 35s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 34s |  |  the patch passed with JDK Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04  |
| +1 :green_heart: |  javadoc  |   0m 32s |  |  the patch passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  spotbugs  |   1m  2s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  23m 29s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  |   1m 17s | [/patch-unit-hadoop-common-project_hadoop-registry.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/1/artifact/out/patch-unit-hadoop-common-project_hadoop-registry.txt) |  hadoop-registry in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 48s |  |  The patch does not generate ASF License warnings.  |
|  |   | 183m  5s |  |  |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.registry.server.dns.TestSecureRegistryDNS |
|   | hadoop.registry.server.dns.TestRegistryDNS |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/3935 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell |
| uname | Linux 142ffc858285 4.15.0-163-generic #171-Ubuntu SMP Fri Nov 5 11:55:11 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / e20271e894a20a95a7988efb4c9db766e92c062a |
| Default Java | Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/1/testReport/ |
| Max. process+thread count | 520 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-registry U: hadoop-common-project/hadoop-registry |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/1/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

steveloughran commented:
lots of test failures there. and is there a way to add a test for the new feature?

eubnara commented:
@steveloughran 
Sorry, I found some bugs in my codes. I'm working on it.

eubnara commented:
It fixes:

- add missing chained CNAME records
  - e.g. when querying `repo.maven.apache.org.` for A record, chained CNAME records should be in answer section.
```
❯ dig @8.8.8.8 repo.maven.apache.org

; <<>> DiG 9.16.1-Ubuntu <<>> @8.8.8.8 repo.maven.apache.org
; (1 server found)
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 65465
;; flags: qr rd ra; QUERY: 1, ANSWER: 4, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 512
;; QUESTION SECTION:
;repo.maven.apache.org.		IN	A

;; ANSWER SECTION:
repo.maven.apache.org.	20102	IN	CNAME	repo.apache.maven.org.
repo.apache.maven.org.	21178	IN	CNAME	maven.map.fastly.net.
maven.map.fastly.net.	4	IN	A	199.232.192.215
maven.map.fastly.net.	4	IN	A	199.232.196.215

;; Query time: 148 msec
;; SERVER: 8.8.8.8#53(8.8.8.8)
;; WHEN: 목  1월 27 12:18:39 KST 2022
;; MSG SIZE  rcvd: 148
```

- distinguish between NXDOMAIN and NOERROR + empty answer from upstream DNS server
  - e.g. when querying `repo.maven.apache.org.` for AAAA record, it should respond `NOERROR` with empty answer not `NXDOMAIN` because A record exists even though AAAA record does not exist.
  - Please note that `status: NOERROR` in result of `dig` command.

```
❯ dig @8.8.8.8 repo.maven.apache.org AAAA

; <<>> DiG 9.16.1-Ubuntu <<>> @8.8.8.8 repo.maven.apache.org AAAA
; (1 server found)
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 5782
;; flags: qr rd ra; QUERY: 1, ANSWER: 2, AUTHORITY: 1, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 512
;; QUESTION SECTION:
;repo.maven.apache.org.		IN	AAAA

;; ANSWER SECTION:
repo.maven.apache.org.	21161	IN	CNAME	repo.apache.maven.org.
repo.apache.maven.org.	20107	IN	CNAME	maven.map.fastly.net.

;; AUTHORITY SECTION:
fastly.net.		30	IN	SOA	ns1.fastly.net. hostmaster.fastly.com. 2017052201 3600 600 604800 30

;; Query time: 164 msec
;; SERVER: 8.8.8.8#53(8.8.8.8)
;; WHEN: 목  1월 27 12:20:47 KST 2022
;; MSG SIZE  rcvd: 177
```

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 57s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  34m 53s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  24m  3s |  |  trunk passed with JDK Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04  |
| +1 :green_heart: |  compile  |  20m 33s |  |  trunk passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  checkstyle  |   0m 30s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 37s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   0m 35s |  |  trunk passed with JDK Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04  |
| +1 :green_heart: |  javadoc  |   0m 33s |  |  trunk passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  spotbugs  |   0m 53s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  23m 33s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| -1 :x: |  mvninstall  |   0m 18s | [/patch-mvninstall-hadoop-common-project_hadoop-registry.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/2/artifact/out/patch-mvninstall-hadoop-common-project_hadoop-registry.txt) |  hadoop-registry in the patch failed.  |
| -1 :x: |  compile  |   1m 57s | [/patch-compile-root-jdkUbuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/2/artifact/out/patch-compile-root-jdkUbuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04.txt) |  root in the patch failed with JDK Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04.  |
| -1 :x: |  javac  |   1m 57s | [/patch-compile-root-jdkUbuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/2/artifact/out/patch-compile-root-jdkUbuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04.txt) |  root in the patch failed with JDK Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04.  |
| -1 :x: |  compile  |   1m 39s | [/patch-compile-root-jdkPrivateBuild-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/2/artifact/out/patch-compile-root-jdkPrivateBuild-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07.txt) |  root in the patch failed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07.  |
| -1 :x: |  javac  |   1m 39s | [/patch-compile-root-jdkPrivateBuild-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/2/artifact/out/patch-compile-root-jdkPrivateBuild-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07.txt) |  root in the patch failed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07.  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -0 :warning: |  checkstyle  |   0m 14s | [/results-checkstyle-hadoop-common-project_hadoop-registry.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/2/artifact/out/results-checkstyle-hadoop-common-project_hadoop-registry.txt) |  hadoop-common-project/hadoop-registry: The patch generated 13 new + 1 unchanged - 0 fixed = 14 total (was 1)  |
| -1 :x: |  mvnsite  |   0m 19s | [/patch-mvnsite-hadoop-common-project_hadoop-registry.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/2/artifact/out/patch-mvnsite-hadoop-common-project_hadoop-registry.txt) |  hadoop-registry in the patch failed.  |
| +1 :green_heart: |  javadoc  |   0m 19s |  |  the patch passed with JDK Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04  |
| +1 :green_heart: |  javadoc  |   0m 17s |  |  the patch passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| -1 :x: |  spotbugs  |   0m 17s | [/patch-spotbugs-hadoop-common-project_hadoop-registry.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/2/artifact/out/patch-spotbugs-hadoop-common-project_hadoop-registry.txt) |  hadoop-registry in the patch failed.  |
| -1 :x: |  shadedclient  |   9m 24s |  |  patch has errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  |   0m 19s | [/patch-unit-hadoop-common-project_hadoop-registry.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/2/artifact/out/patch-unit-hadoop-common-project_hadoop-registry.txt) |  hadoop-registry in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 28s |  |  The patch does not generate ASF License warnings.  |
|  |   | 122m  1s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/3935 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell |
| uname | Linux 09351c073828 4.15.0-163-generic #171-Ubuntu SMP Fri Nov 5 11:55:11 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / c6094e25563b29058fff0280052ed91024dbdf61 |
| Default Java | Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/2/testReport/ |
| Max. process+thread count | 522 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-registry U: hadoop-common-project/hadoop-registry |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/2/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 37s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 1 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  33m 38s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  24m 37s |  |  trunk passed with JDK Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04  |
| +1 :green_heart: |  compile  |  21m 11s |  |  trunk passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  checkstyle  |   0m 40s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 48s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   0m 47s |  |  trunk passed with JDK Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04  |
| +1 :green_heart: |  javadoc  |   0m 44s |  |  trunk passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  spotbugs  |   1m  7s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  21m 10s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 23s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  23m 42s |  |  the patch passed with JDK Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04  |
| +1 :green_heart: |  javac  |  23m 42s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  20m 58s |  |  the patch passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  javac  |  20m 58s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -0 :warning: |  checkstyle  |   0m 38s | [/results-checkstyle-hadoop-common-project_hadoop-registry.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/3/artifact/out/results-checkstyle-hadoop-common-project_hadoop-registry.txt) |  hadoop-common-project/hadoop-registry: The patch generated 14 new + 1 unchanged - 0 fixed = 15 total (was 1)  |
| +1 :green_heart: |  mvnsite  |   0m 46s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 44s |  |  the patch passed with JDK Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04  |
| +1 :green_heart: |  javadoc  |   0m 43s |  |  the patch passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  spotbugs  |   1m 15s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  21m 22s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   1m 26s |  |  hadoop-registry in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 59s |  |  The patch does not generate ASF License warnings.  |
|  |   | 181m 29s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/3935 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell |
| uname | Linux 59b770f058c4 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / dad60e77f1ade2e1eaa6b03cbf691b67bfb2576b |
| Default Java | Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/3/testReport/ |
| Max. process+thread count | 549 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-registry U: hadoop-common-project/hadoop-registry |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/3/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

eubnara committed:
HADOOP-18087 support both CNAME and DNAME chaining

- There is no way to figure out the type of alias from dnsjava's lookup. As a workaround, query CNAME first and then DNAME.
- dnsjava use "dnsjava.lookup.max_iterations" property in order to limit follow chaining. Default 16. Therefore, I think "iterations" var. in remoteLookup() is not necessary.

hadoop-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 49s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 1 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  35m  6s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  26m 23s |  |  trunk passed with JDK Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04  |
| +1 :green_heart: |  compile  |  22m 27s |  |  trunk passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  checkstyle  |   0m 38s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 43s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   0m 43s |  |  trunk passed with JDK Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04  |
| +1 :green_heart: |  javadoc  |   0m 40s |  |  trunk passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  spotbugs  |   1m  3s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  24m 25s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 24s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  22m 48s |  |  the patch passed with JDK Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04  |
| +1 :green_heart: |  javac  |  22m 48s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  20m 24s |  |  the patch passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  javac  |  20m 24s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -0 :warning: |  checkstyle  |   0m 37s | [/results-checkstyle-hadoop-common-project_hadoop-registry.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/4/artifact/out/results-checkstyle-hadoop-common-project_hadoop-registry.txt) |  hadoop-common-project/hadoop-registry: The patch generated 15 new + 1 unchanged - 0 fixed = 16 total (was 1)  |
| +1 :green_heart: |  mvnsite  |   0m 45s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 36s |  |  the patch passed with JDK Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04  |
| +1 :green_heart: |  javadoc  |   0m 38s |  |  the patch passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  spotbugs  |   1m 13s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  21m 40s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   1m 22s |  |  hadoop-registry in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 54s |  |  The patch does not generate ASF License warnings.  |
|  |   | 186m 59s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/3935 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell |
| uname | Linux 005f79ad197d 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / cf1e3776b25eb889a43bb736cfb0c15f83996549 |
| Default Java | Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/4/testReport/ |
| Max. process+thread count | 549 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-registry U: hadoop-common-project/hadoop-registry |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/4/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

eubnara commented:
@steveloughran 
Could you remove a stale tag not to close this PR, please?

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 42s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 1 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  22m  1s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   8m 45s |  |  trunk passed with JDK Ubuntu-21.0.7+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  compile  |   8m 45s |  |  trunk passed with JDK Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  checkstyle  |   0m 25s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 27s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   0m 27s |  |  trunk passed with JDK Ubuntu-21.0.7+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   0m 24s |  |  trunk passed with JDK Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04  |
| -1 :x: |  spotbugs  |   0m 36s | [/branch-spotbugs-hadoop-common-project_hadoop-registry-warnings.html](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/5/artifact/out/branch-spotbugs-hadoop-common-project_hadoop-registry-warnings.html) |  hadoop-common-project/hadoop-registry in trunk has 19 extant spotbugs warnings.  |
| +1 :green_heart: |  shadedclient  |  13m 59s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| -1 :x: |  mvninstall  |   0m 12s | [/patch-mvninstall-hadoop-common-project_hadoop-registry.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/5/artifact/out/patch-mvninstall-hadoop-common-project_hadoop-registry.txt) |  hadoop-registry in the patch failed.  |
| -1 :x: |  compile  |   0m 48s | [/patch-compile-root-jdkUbuntu-21.0.7+6-Ubuntu-0ubuntu120.04.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/5/artifact/out/patch-compile-root-jdkUbuntu-21.0.7+6-Ubuntu-0ubuntu120.04.txt) |  root in the patch failed with JDK Ubuntu-21.0.7+6-Ubuntu-0ubuntu120.04.  |
| -1 :x: |  javac  |   0m 48s | [/patch-compile-root-jdkUbuntu-21.0.7+6-Ubuntu-0ubuntu120.04.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/5/artifact/out/patch-compile-root-jdkUbuntu-21.0.7+6-Ubuntu-0ubuntu120.04.txt) |  root in the patch failed with JDK Ubuntu-21.0.7+6-Ubuntu-0ubuntu120.04.  |
| -1 :x: |  compile  |   0m 50s | [/patch-compile-root-jdkUbuntu-17.0.15+6-Ubuntu-0ubuntu120.04.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/5/artifact/out/patch-compile-root-jdkUbuntu-17.0.15+6-Ubuntu-0ubuntu120.04.txt) |  root in the patch failed with JDK Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04.  |
| -1 :x: |  javac  |   0m 50s | [/patch-compile-root-jdkUbuntu-17.0.15+6-Ubuntu-0ubuntu120.04.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/5/artifact/out/patch-compile-root-jdkUbuntu-17.0.15+6-Ubuntu-0ubuntu120.04.txt) |  root in the patch failed with JDK Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04.  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -0 :warning: |  checkstyle  |   0m  9s | [/results-checkstyle-hadoop-common-project_hadoop-registry.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/5/artifact/out/results-checkstyle-hadoop-common-project_hadoop-registry.txt) |  hadoop-common-project/hadoop-registry: The patch generated 15 new + 1 unchanged - 0 fixed = 16 total (was 1)  |
| -1 :x: |  mvnsite  |   0m 13s | [/patch-mvnsite-hadoop-common-project_hadoop-registry.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/5/artifact/out/patch-mvnsite-hadoop-common-project_hadoop-registry.txt) |  hadoop-registry in the patch failed.  |
| -1 :x: |  javadoc  |   0m 14s | [/results-javadoc-javadoc-hadoop-common-project_hadoop-registry-jdkUbuntu-21.0.7+6-Ubuntu-0ubuntu120.04.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/5/artifact/out/results-javadoc-javadoc-hadoop-common-project_hadoop-registry-jdkUbuntu-21.0.7+6-Ubuntu-0ubuntu120.04.txt) |  hadoop-common-project_hadoop-registry-jdkUbuntu-21.0.7+6-Ubuntu-0ubuntu120.04 with JDK Ubuntu-21.0.7+6-Ubuntu-0ubuntu120.04 generated 5 new + 136 unchanged - 0 fixed = 141 total (was 136)  |
| -1 :x: |  javadoc  |   0m 12s | [/results-javadoc-javadoc-hadoop-common-project_hadoop-registry-jdkUbuntu-17.0.15+6-Ubuntu-0ubuntu120.04.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/5/artifact/out/results-javadoc-javadoc-hadoop-common-project_hadoop-registry-jdkUbuntu-17.0.15+6-Ubuntu-0ubuntu120.04.txt) |  hadoop-common-project_hadoop-registry-jdkUbuntu-17.0.15+6-Ubuntu-0ubuntu120.04 with JDK Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04 generated 5 new + 129 unchanged - 0 fixed = 134 total (was 129)  |
| -1 :x: |  spotbugs  |   0m 12s | [/patch-spotbugs-hadoop-common-project_hadoop-registry.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/5/artifact/out/patch-spotbugs-hadoop-common-project_hadoop-registry.txt) |  hadoop-registry in the patch failed.  |
| -1 :x: |  shadedclient  |   6m  9s |  |  patch has errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  |   0m 14s | [/patch-unit-hadoop-common-project_hadoop-registry.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/5/artifact/out/patch-unit-hadoop-common-project_hadoop-registry.txt) |  hadoop-registry in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 18s |  |  The patch does not generate ASF License warnings.  |
|  |   |  66m 33s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.52 ServerAPI=1.52 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/3935 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets |
| uname | Linux 899eb9c5caf6 5.15.0-156-generic #166-Ubuntu SMP Sat Aug 9 00:02:46 UTC 2025 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 80e5c2a3617ae738a411855b187a8914ea53798a |
| Default Java | Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04 |
| Multi-JDK versions | /usr/lib/jvm/java-21-openjdk-amd64:Ubuntu-21.0.7+6-Ubuntu-0ubuntu120.04 /usr/lib/jvm/java-17-openjdk-amd64:Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/5/testReport/ |
| Max. process+thread count | 613 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-registry U: hadoop-common-project/hadoop-registry |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/5/console |
| versions | git=2.25.1 maven=3.9.11 spotbugs=4.9.7 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

eubnara committed:
HADOOP-18087. fix bugs on TestRegistryDNS.java

eubnara commented:
I made new PR at https://github.com/apache/hadoop/pull/8091

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 42s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 1 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  22m 33s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   9m 52s |  |  trunk passed with JDK Ubuntu-21.0.7+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  compile  |   9m 57s |  |  trunk passed with JDK Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  checkstyle  |   0m 20s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 25s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   0m 26s |  |  trunk passed with JDK Ubuntu-21.0.7+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   0m 24s |  |  trunk passed with JDK Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04  |
| -1 :x: |  spotbugs  |   0m 36s | [/branch-spotbugs-hadoop-common-project_hadoop-registry-warnings.html](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/6/artifact/out/branch-spotbugs-hadoop-common-project_hadoop-registry-warnings.html) |  hadoop-common-project/hadoop-registry in trunk has 19 extant spotbugs warnings.  |
| +1 :green_heart: |  shadedclient  |  15m 10s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 11s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   9m 17s |  |  the patch passed with JDK Ubuntu-21.0.7+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javac  |   9m 17s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  10m  2s |  |  the patch passed with JDK Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javac  |  10m  2s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -0 :warning: |  checkstyle  |   0m 18s | [/results-checkstyle-hadoop-common-project_hadoop-registry.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/6/artifact/out/results-checkstyle-hadoop-common-project_hadoop-registry.txt) |  hadoop-common-project/hadoop-registry: The patch generated 15 new + 1 unchanged - 0 fixed = 16 total (was 1)  |
| +1 :green_heart: |  mvnsite  |   0m 23s |  |  the patch passed  |
| -1 :x: |  javadoc  |   0m 22s | [/results-javadoc-javadoc-hadoop-common-project_hadoop-registry-jdkUbuntu-21.0.7+6-Ubuntu-0ubuntu120.04.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/6/artifact/out/results-javadoc-javadoc-hadoop-common-project_hadoop-registry-jdkUbuntu-21.0.7+6-Ubuntu-0ubuntu120.04.txt) |  hadoop-common-project_hadoop-registry-jdkUbuntu-21.0.7+6-Ubuntu-0ubuntu120.04 with JDK Ubuntu-21.0.7+6-Ubuntu-0ubuntu120.04 generated 5 new + 136 unchanged - 0 fixed = 141 total (was 136)  |
| -1 :x: |  javadoc  |   0m 21s | [/results-javadoc-javadoc-hadoop-common-project_hadoop-registry-jdkUbuntu-17.0.15+6-Ubuntu-0ubuntu120.04.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/6/artifact/out/results-javadoc-javadoc-hadoop-common-project_hadoop-registry-jdkUbuntu-17.0.15+6-Ubuntu-0ubuntu120.04.txt) |  hadoop-common-project_hadoop-registry-jdkUbuntu-17.0.15+6-Ubuntu-0ubuntu120.04 with JDK Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04 generated 5 new + 129 unchanged - 0 fixed = 134 total (was 129)  |
| -1 :x: |  spotbugs  |   0m 37s | [/new-spotbugs-hadoop-common-project_hadoop-registry.html](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/6/artifact/out/new-spotbugs-hadoop-common-project_hadoop-registry.html) |  hadoop-common-project/hadoop-registry generated 1 new + 17 unchanged - 2 fixed = 18 total (was 19)  |
| +1 :green_heart: |  shadedclient  |  14m 57s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 49s |  |  hadoop-registry in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 31s |  |  The patch does not generate ASF License warnings.  |
|  |   | 101m 24s |  |  |


| Reason | Tests |
|-------:|:------|
| SpotBugs | module:hadoop-common-project/hadoop-registry |
|  |  Unknown bug pattern DCN_NULLPOINTER_EXCEPTION in org.apache.hadoop.registry.server.dns.RegistryDNS.getRecords(Name, int)  At RegistryDNS.java:org.apache.hadoop.registry.server.dns.RegistryDNS.getRecords(Name, int)  At RegistryDNS.java:[line 1190] |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.52 ServerAPI=1.52 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/6/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/3935 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets |
| uname | Linux b9f182d68fa9 5.15.0-156-generic #166-Ubuntu SMP Sat Aug 9 00:02:46 UTC 2025 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 683b944943662ca2af3a5953ecd7428768ec4173 |
| Default Java | Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04 |
| Multi-JDK versions | /usr/lib/jvm/java-21-openjdk-amd64:Ubuntu-21.0.7+6-Ubuntu-0ubuntu120.04 /usr/lib/jvm/java-17-openjdk-amd64:Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/6/testReport/ |
| Max. process+thread count | 613 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-registry U: hadoop-common-project/hadoop-registry |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/6/console |
| versions | git=2.25.1 maven=3.9.11 spotbugs=4.9.7 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 44s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 1 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  23m 17s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   9m 55s |  |  trunk passed with JDK Ubuntu-21.0.7+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  compile  |   9m 19s |  |  trunk passed with JDK Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  checkstyle  |   0m 19s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 23s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   0m 26s |  |  trunk passed with JDK Ubuntu-21.0.7+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   0m 22s |  |  trunk passed with JDK Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04  |
| -1 :x: |  spotbugs  |   0m 36s | [/branch-spotbugs-hadoop-common-project_hadoop-registry-warnings.html](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/7/artifact/out/branch-spotbugs-hadoop-common-project_hadoop-registry-warnings.html) |  hadoop-common-project/hadoop-registry in trunk has 19 extant spotbugs warnings.  |
| +1 :green_heart: |  shadedclient  |  15m 17s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 12s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   9m 10s |  |  the patch passed with JDK Ubuntu-21.0.7+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javac  |   9m 10s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   9m 26s |  |  the patch passed with JDK Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javac  |   9m 26s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -0 :warning: |  checkstyle  |   0m 16s | [/results-checkstyle-hadoop-common-project_hadoop-registry.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/7/artifact/out/results-checkstyle-hadoop-common-project_hadoop-registry.txt) |  hadoop-common-project/hadoop-registry: The patch generated 15 new + 1 unchanged - 0 fixed = 16 total (was 1)  |
| +1 :green_heart: |  mvnsite  |   0m 22s |  |  the patch passed  |
| -1 :x: |  javadoc  |   0m 20s | [/results-javadoc-javadoc-hadoop-common-project_hadoop-registry-jdkUbuntu-21.0.7+6-Ubuntu-0ubuntu120.04.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/7/artifact/out/results-javadoc-javadoc-hadoop-common-project_hadoop-registry-jdkUbuntu-21.0.7+6-Ubuntu-0ubuntu120.04.txt) |  hadoop-common-project_hadoop-registry-jdkUbuntu-21.0.7+6-Ubuntu-0ubuntu120.04 with JDK Ubuntu-21.0.7+6-Ubuntu-0ubuntu120.04 generated 5 new + 136 unchanged - 0 fixed = 141 total (was 136)  |
| -1 :x: |  javadoc  |   0m 22s | [/results-javadoc-javadoc-hadoop-common-project_hadoop-registry-jdkUbuntu-17.0.15+6-Ubuntu-0ubuntu120.04.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/7/artifact/out/results-javadoc-javadoc-hadoop-common-project_hadoop-registry-jdkUbuntu-17.0.15+6-Ubuntu-0ubuntu120.04.txt) |  hadoop-common-project_hadoop-registry-jdkUbuntu-17.0.15+6-Ubuntu-0ubuntu120.04 with JDK Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04 generated 5 new + 129 unchanged - 0 fixed = 134 total (was 129)  |
| -1 :x: |  spotbugs  |   0m 41s | [/new-spotbugs-hadoop-common-project_hadoop-registry.html](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/7/artifact/out/new-spotbugs-hadoop-common-project_hadoop-registry.html) |  hadoop-common-project/hadoop-registry generated 1 new + 17 unchanged - 2 fixed = 18 total (was 19)  |
| +1 :green_heart: |  shadedclient  |  15m  2s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 53s |  |  hadoop-registry in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 34s |  |  The patch does not generate ASF License warnings.  |
|  |   | 101m  0s |  |  |


| Reason | Tests |
|-------:|:------|
| SpotBugs | module:hadoop-common-project/hadoop-registry |
|  |  Unknown bug pattern DCN_NULLPOINTER_EXCEPTION in org.apache.hadoop.registry.server.dns.RegistryDNS.getRecords(Name, int)  At RegistryDNS.java:org.apache.hadoop.registry.server.dns.RegistryDNS.getRecords(Name, int)  At RegistryDNS.java:[line 1190] |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.52 ServerAPI=1.52 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/7/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/3935 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets |
| uname | Linux 9cd6dd3fc026 5.15.0-156-generic #166-Ubuntu SMP Sat Aug 9 00:02:46 UTC 2025 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 683b944943662ca2af3a5953ecd7428768ec4173 |
| Default Java | Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04 |
| Multi-JDK versions | /usr/lib/jvm/java-21-openjdk-amd64:Ubuntu-21.0.7+6-Ubuntu-0ubuntu120.04 /usr/lib/jvm/java-17-openjdk-amd64:Ubuntu-17.0.15+6-Ubuntu-0ubuntu120.04 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/7/testReport/ |
| Max. process+thread count | 610 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-registry U: hadoop-common-project/hadoop-registry |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3935/7/console |
| versions | git=2.25.1 maven=3.9.11 spotbugs=4.9.7 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.
